lib/Dialect/TritonGPU/Transforms/Pipeliner/MMAv5PipelineUtility.cpp

```diff
@@ -0,0 +1,301 @@
+#include "triton/Dialect/TritonGPU/Transforms/MMAv5PipelineUtility.h"
```

Comment:
do you know how it will interact with the refactor from @pawelszczerbuk https://github.com/triton-lang/triton/pull/6256?

Reply:
I read over the refactor, and this can definitely reuse the hoistTMEMAlloc pass which simplifies a lot of analysis. However, the rest of the logic on top of MMAInfo has to be rewritten on top of the new utilities. It's not clear how much can be reused yet. I am planning to fast-follow Pawel's refactor with a rewrite so that the old MMAv5 analysis can be deleted.